tests/generative/variable_strategies.py

```diff
@@ -183,12 +200,84 @@ def subtract(draw, type_, frame):
     def multiply(draw, type_, frame):
         return draw(binary_operation(type_, frame, Function.Multiply))
 
+    @st.composite
+    def date_add_years(draw, type_, frame):
+        return draw(
+            binary_operation_with_types(type_, int, frame, Function.DateAddYears)
+        )
+
+    @st.composite
+    def date_add_months(draw, type_, frame):
+        return draw(
+            binary_operation_with_types(type_, int, frame, Function.DateAddMonths)
+        )
+
+    @st.composite
+    def date_add_days(draw, type_, frame):
+        return draw(
+            binary_operation_with_types(type_, int, frame, Function.DateAddDays)
+        )
+
+    @st.composite
+    def date_difference_in_years(draw, type_, frame):
+        return draw(
+            binary_operation(datetime.date, frame, Function.DateDifferenceInYears)
+        )
+
+    @st.composite
+    def date_difference_in_months(draw, type_, frame):
+        return draw(
+            binary_operation(datetime.date, frame, Function.DateDifferenceInMonths)
+        )
+
+    @st.composite
+    def date_difference_in_days(draw, type_, frame):
+        return draw(
+            binary_operation(datetime.date, frame, Function.DateDifferenceInDays)
+        )
+
     @st.composite
     def binary_operation(draw, type_, frame, operator_func):
         # A strategy for operations that take lhs and rhs arguments of the
         # same type
-        lhs = draw(series(type_, draw(one_row_per_patient_frame_or(frame))))
-        rhs = draw(series(type_, draw(one_row_per_patient_frame_or(frame))))
+        return draw(binary_operation_with_types(type_, type_, frame, operator_func))
+
+    @st.composite
+    def binary_operation_with_types(draw, lhs_type, rhs_type, frame, operator_func):
+        # A strategy for operations that take lhs and rhs arguments with specified lhs
+        # and rhs types (which may be different)
+
+        # A binary operation has 2 inputs, which are
+        # 1) A series drawn from the specified frame
+        # 2) one of:
+        #    a) A series drawn from the specified frame
+        #    b) A series drawn from a one-row-per-patient-frame
```

Comment:
Maybe "any one-row-per-patient frame" to make it clear that it can be different in the case where such a thing is passed in?